.github/workflows/ci.yml

```diff
@@ -7,6 +7,33 @@ on:
     branches: [ main ]
 
 jobs:
+  cppcheck_analyser:
+    name: static check with cppcheck
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        submodules: 'recursive'
+    - name: setup
+      env:
+        CC: /usr/bin/gcc-10
+        CXX: /usr/bin/g++-10
+      run: |
+        sudo -E ./ci/setup_cmake.sh
+        sudo -E ./ci/setup_ci_environment.sh
+        sudo apt-get -y install cmake
+    - name: build and run cppcheck
+      run: |
+        cd ..
+        git clone https://github.com/danmar/cppcheck.git --branch 2.12.x
+        cd cppcheck
+        mkdir build
+        cd build
+        cmake ..
+        cmake --build .
+        chmod +x bin/cppcheck
+        cd ../../opentelemetry-cpp
+        ../cppcheck/build/bin/cppcheck -ithird_party/ --error-exitcode=1 .
```

Comment:
Should we have this CI similar to that for `format` : https://github.com/open-telemetry/opentelemetry-cpp/blob/11d5d9e0d8fd8ba876c8994714cc2647479b6574/.github/workflows/ci.yml#L678-L686

- Create script to install a particular version of cppcheck passed as argument.
- And invoke the cppcheck through `./do_ci.sh cppcheck`

This will also allow developers to invoke static analysis locally.

Reply:
And should `test_common` also be ignored here?